compiler/src/dotty/tools/dotc/config/SourceVersion.scala

```diff
@@ -43,7 +43,7 @@ enum SourceVersion:
   def enablesClauseInterleaving = isAtLeast(`3.6`)
   def enablesNewGivens = isAtLeast(`3.6`)
   def enablesNamedTuples = isAtLeast(`3.7`)
-  def enablesBetterFors(using Context) = isAtLeast(`3.7`) && isPreviewEnabled
+  def enablesBetterFors = isAtLeast(`3.8`)
```

Comment:
I wonder should we also enable it under `-source:3.7`, `-preview` - it makes sense to me as well
```suggestion
  def enablesBetterFors(using Context) = isAtLeast(`3.8`) || ( isAtLeast(`3.7`) && isPreviewEnabled)```

Reply:
makes sense to me, too